### PR TITLE
Add support for bazel buildsystem to physics3

### PR DESCRIPTION
**Pull request by mjcarroll**

### CODE REVIEW

File: dartsim/BUILD.bazel [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+        IGNITION_ROOT + "ign_math/eigen3",
+        IGNITION_ROOT + "ign_physics",
+        IGNITION_ROOT + "ign_plugin/register",
+        IGNITION_ROOT + "sdformat",
+    ],
+)
+
+cc_binary(
+    name = "libignition-physics-dartsim-plugin.so",
+    srcs = [":dartsim"],
```
Comment (mahiuchun) [inline dartsim/BUILD.bazel:76]: use deps instead of srcs?

File: tpe/BUILD.bazel [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+        ":tpelib",
+        IGNITION_ROOT + "ign_math/eigen3",
+        IGNITION_ROOT + "ign_physics",
+        IGNITION_ROOT + "ign_plugin/register",
+    ],
+)
+
+cc_binary(
+    name = "libignition-physics-tpe-plugin.so",
+    srcs = [":plugin"],
```
Comment (mahiuchun) [inline tpe/BUILD.bazel:96]: use deps instead of srcs?

File: test/PhysicsPluginsList.hh [hunk 3]
```diff
@@ -21,18 +21,54 @@
...
 
 #ifndef tpe_plugin_LIB
-#define tpe_plugin_LIB ""
+#define tpe_plugin_LIB "__main__/ign_physics/tpe/libignition-physics-tpe-plugin.so"
 #endif
 
+/////////////////////////////////////////////////
+std::string resolveLibrary(const std::string &_path)
+{
+  auto it = _path.find("__main__");
```
Comment (mahiuchun) [inline test/PhysicsPluginsList.hh:42]: turns out our internal version of Bazel does not use `__main__` in its paths. could we only check the existence of `TEST_SRCDIR`?

**Review by mahiuchun — APPROVED** [review]

**Review by chapulina — CHANGES_REQUESTED** [review]
CI is broken

**Comment by chapulina** [comment]
I believe this PR is outdated and not being worked on. Please reopen if this starts being worked on again.

**Pull request closed without merging**